app_flutter/lib/task_box.dart

```diff
@@ -215,38 +212,119 @@ class TaskOverlayContents extends StatelessWidget {
+    @required this.task,
+    @required this.taskStatus,
+  })  : assert(parentContext != null),
+        assert(buildState != null),
+        assert(task != null),
+        super(key: key);
+
+  /// The parent context that has the size of the whole screen
+  final BuildContext parentContext;
+
+  /// A reference to the [FlutterBuildState] for performing operations on this [Task].
+  final FlutterBuildState buildState;
+
+  /// The [Task] to display in the overlay
+  final Task task;
+
+  /// [Task.status] modified to take into account [Task.attempts] to create
+  /// a more descriptive status.
+  final String taskStatus;
+
+  @visibleForTesting
+  static const String rerunErrorMessage = 'Failed to rerun task.';
+  @visibleForTesting
+  static const String rerunSuccessMessage =
+      'Devicelab is rerunning the task. This can take a minute to propagate.';
+  @visibleForTesting
+  static const Duration rerunSnackbarDuration = Duration(seconds: 15);
+
+  /// A lookup table to define the [Icon] for this [taskStatus].
+  static const Map<String, Icon> statusIcon = <String, Icon>{
+    TaskBox.statusFailed: Icon(Icons.clear, color: Colors.red, size: 32),
+    TaskBox.statusNew: Icon(Icons.new_releases, color: Colors.blue, size: 32),
+    TaskBox.statusInProgress:
+        Icon(Icons.autorenew, color: Colors.blue, size: 32),
+    TaskBox.statusSucceeded:
+        Icon(Icons.check_circle, color: Colors.green, size: 32),
+    TaskBox.statusSucceededButFlaky: Icon(Icons.check_circle_outline, size: 32),
+    TaskBox.statusUnderperformed:
+        Icon(Icons.new_releases, color: Colors.orange, size: 32),
+    TaskBox.statusUnderperformedInProgress:
+        Icon(Icons.autorenew, color: Colors.orange, size: 32),
+  };
+
+  @override
+  Widget build(BuildContext context) {
+    return Card(
+      child: Column(
+        mainAxisSize: MainAxisSize.max,
+        children: <Widget>[
+          ListTile(
+            leading:
+                Tooltip(message: taskStatus, child: statusIcon[taskStatus]),
+            title: Text(task.name),
+            subtitle: Text(
+                'Attempts: ${task.attempts}\nDuration: ${task.endTimestamp - task.startTimestamp} seconds\nAgent: ${task.reservedForAgentId}'),
+          ),
+          ButtonBar(
+            children: <Widget>[
+              IconButton(
+                icon: const Icon(Icons.receipt),
+                onPressed: () {
+                  // TODO(chillers): Open log in new window. https://github.com/flutter/cocoon/issues/436
+                },
+              ),
+              if (isDevicelab(task))
+                ProgressButton(
+                  defaultWidget: const Text('Rerun task'),
+                  progressWidget: const CircularProgressIndicator(),
+                  width: 120,
+                  height: 50,
+                  onPressed: () async {
+                    final bool success = await buildState.rerunTask(task);
+                    return () {
```

Comment:
Why are you returning a callback here?

Reply:
It was in the example, and I have never done it before :)

Removed since it didn't go anywhere.